### PR TITLE
Start an Application object

### PR DESCRIPTION
**Pull request by whit537**
This creates a new `Application` object that subsumes `Website` and should eventually eat all of `wireup`(!). ~~Immediate next step is to make an `Email` mixin for it: #4346.~~ We need to make this available to payday and models before we can work on email: #4350.

### Todo

- [x] change base to `master`

### CODE REVIEW

File: www/~/%username/emails/verify.html.spt [hunk 1]
```diff
@@ -55,33 +56,34 @@ suppress_sidebar = True
...
-            {% elif result == emails.VERIFICATION_FAILED %}
+                        ("<b>%s</b>"|safe) % email_address) }}</p>
+            {% elif result == email.VERIFICATION_FAILED %}
                 <h1>{{ _("Failure") }}</h1>
                 <p>{{ _("The verification code for {0} is bad.",
-                        ("<b>%s</b>"|safe) % email) }}</p>
+                        ("<b>%s</b>"|safe) % email_address) }}</p>
             {% endif %}
-            <p data-email="{{ email }}"><button class="resend">{{ _("Resend verification email") }}</button></p>
+            <p>Result: {{ result }}</p>
```
Comment (whit537) [inline www/~/%username/emails/verify.html.spt]: Stray log line.

**Comment by whit537** [comment]
Rebased, was 9d154116aa75093d148a724acf2c7e00eef2f0a3.

**Comment by whit537** [comment]
Okay! This PR is weighing in at +/-440 ... not as bad as I thought. :)

**Comment by whit537** [comment]
Erm, well, now up to 516. 😁

**Comment by whit537** [comment]
Rebased, was 24c1eb672e566fae56db1ca19ffe82f9277d53f4.

**Comment by whit537** [comment]
This one is up next, @nobodxbodon et al.

**Comment by whit537** [comment]
"cli" means "command line interface"

[Google search](https://www.google.com/search?q=cli) suggests that this is a common usage.

**Comment by nobodxbodon** [comment]
@whit537 thanks. I didn't realize the dequeue_emails/list_email_queue are command line tools. Now it makes sense. Would you like it to be reviewed more in depth by someone else or shall I merge it?

**Comment by whit537** [comment]
Yeah, we added those scripts a month or two ago when we wanted to send out a notification about our new terms of service (#4232).

Rebased again (was d39c8a36bbd5a82ebb7396569065e6996cd290b3) to catch up with recent [bugfix](http://gratipay.slackarchive.io/gratipay/-/1488309786.007464/1488482054.007634/1488480110007570/) [deploys](http://gratipay.slackarchive.io/gratipay/-/1488309786.007464/1488482054.007634/1488481161007633/). I'm ready to go here if you are, @nobodxbodon. Merge away! :-)

**Comment by nobodxbodon** [comment]
I wonder how to restart travis manually? Or is it possible to config travis to retry automatically in case of timeout (no failed cases)?

**Comment by mattbk** [comment]
If you don't have whatever special privileges @whit537 has, one method is to make a small commit and push to jumpstart Travis again.

**Comment by whit537** [comment]
Yeah, there's a little "Restart build" button at upper right. Not sure why you don't have perms ...

**Comment by whit537** [comment]
screen shot 2017-03-03 at 7 43 20 am

**Comment by whit537** [comment]
Restarted ...

**Comment by whit537** [comment]
Green! :-)

**Comment by mattbk** [comment]
> Yeah, there's a little "Restart build" button at upper right. Not sure why you don't have perms ...

On Travis itself? I don't have that.

**Comment by whit537** [comment]
After poking around for a bit in the Travis UI, the closest thing I found to something that might help us here is the "Sync" button for the Gratipay GitHub org. I've gone ahead and clicked that ... did that update perms sufficient to enable the "Restart build" button for yinz?

screen shot 2017-03-03 at 11 17 34 am

**Comment by mattbk** [comment]
Got it. I had to be signed into Travis using GitHub credentials.

**Comment by whit537** [comment]
Ah! That makes sense. :-)

!m @mattbk

**Comment by whit537** [comment]
💃

**Comment by whit537** [comment]
This is confusing. I'm trying to deploy this and getting an error:

```
[gratipay] $ ./deploy.sh 
Already up-to-date.
Traceback (most recent call last):
  File "gratipay/wireup.py", line 11, in <module>
    import aspen
  File "/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/aspen/__init__.py", line 65, in <module>
    import pkg_resources
  File "/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 37, in <module>
    import email.parser
  File "/Users/whit537/personal/gratipay/gratipay.com/gratipay/email.py", line 5, in <module>
    from aspen.simplates.pagination import parse_specline, split_and_escape
  File "/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/aspen/simplates/__init__.py", line 15, in <module>
    from .. import log
ImportError: cannot import name log
[gratipay] $
```

**Comment by whit537** [comment]
```
[gratipay] $ git log | head -n1
commit 796d36e914a44b88e60964416e754c0b473ca163
[gratipay] $
```

**Comment by whit537** [comment]
I ran `make clean env`.

**Comment by whit537** [comment]
796d36e914a44b88e60964416e754c0b473ca163 is the merge of this PR into `master`:

https://github.com/gratipay/gratipay.com/commits/796d36e914a44b88e60964416e754c0b473ca163

**Comment by whit537** [comment]
Moving to #4364.

**Pull request merged**